animations/lib/src/misc/physics_card_drag.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/physics.dart';
+
+class PhysicsCardDragDemo extends StatelessWidget {
+  static const String routeName = '/misc/physics_card';
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(),
+      body: DraggableCard(
+        child: FlutterLogo(
+          size: 128,
+        ),
+      ),
+    );
+  }
+}
+
+/// A draggable card that moves back to [Alignment.center] when it's
+/// released.
+class DraggableCard extends StatefulWidget {
+  final Widget child;
+  DraggableCard({this.child});
+
+  @override
+  _DraggableCardState createState() => _DraggableCardState();
+}
+
+/// A draggable card that moves back to [Alignment.center] when it's
+/// released.
+class _DraggableCardState extends State<DraggableCard>
+    with SingleTickerProviderStateMixin {
+  AnimationController _controller;
+  /// The position the user has dragged the card to. Represented as an
+  /// [Alignment].
+  Alignment _dragAlignment = Alignment.center;
+
+  Animation<Alignment> _animation;
+
+  /// Update the animation so that it runs from the dragged point back to the
+  /// center.
+  void _updateAnimation() {
+    _animation = _controller.drive(
+      AlignmentTween(
+        begin: _dragAlignment,
+        end: Alignment.center,
+      ),
+    );
+  }
+
+  @override
+  void initState() {
+    _controller = AnimationController(vsync: this);
+
+    _controller.addListener(() {
+      setState(() {
+        _dragAlignment = _animation.value;
+      });
+    });
+    _updateAnimation();
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    var size = MediaQuery.of(context).size;
+    return GestureDetector(
+      onPanDown: (details) {
+        _controller.stop();
+      },
+      onPanUpdate: (details) {
+        setState(() {
+          _dragAlignment += Alignment(
+            details.delta.dx / (size.width / 2),
+            details.delta.dy / (size.height / 2),
+          );
+        });
+      },
+      onPanEnd: (details) {
+        _updateAnimation();
```

Comment:
I had a good bit of difficulty at first parsing out what's happening here. That's more to do with the complexity of the effect you're trying to achieve than anything to do with your code, but anything that could simplify the code for readers is probably a good idea.

For example, could `_updateAnimation` be changed to accept a velocity and do the work of creating a SpringDescription/Simulation and updating the controller? In `initState` it could be called with a zero velocity, and in `onPanEnd` with a real one.

That causes the code to do a little unnecessary work during the first call in `initState`, but it also moves all the work of setting up the animation and controller to a single, easy-to-read method.

Reply:
For now, I moved the logic from `onDragEnd` into a new method, `runAnimation`. I don't see how to combine `runAnimation` and `updateAnimation`.